rfc/0002/README.md

```diff
@@ -0,0 +1,169 @@
+
+Proposal
+--------
+
+The proposal below might take a couple of read-throughs; I've also added a concrete example afterward that might help.
+
+1. The AUTHOR writes their application expecting NNF Storage at specific locations. For each storage requirement, they define:
+    1. a unique name for the storage which can be referenced in the 'container' directive
+    2. the expected storage types; if necessary
+    3. the required mount path or mount path prefix
+    4. other constraints or storage requirements (e.g. minimum capacity)
+2. The AUTHOR works with the ADMINISTRATOR to define:
+    1. a unique name for the program to be referred by USER
+    2. the pod specification for executing their program
+    3. the NNF storage requirements described above. 
+3. The ADMINISTRATOR creates a corresponding _NNF Container Profile_ custom kubernetes resource
+4. The USER who desires to use the application works with the AUTHOR and the related Container Profile to understand the storage requirements.
+5. The USER submits a WLM job with the #DW container fields populated
+6. WLM runs the job and drives the job through the following stages...
+    1. Proposal: RABBIT validates the #DW container directive by comparing the supplied values to what is listed in the Container Profile. If the USER fails to meet the requirements, the job fails. 
+    2. Pre-run: RABBIT software will:
+        1. create a config map reflecting the storage requirements and any runtime parameters
+        2. duplicate the pod specification from the Container Profile and patches the necessary Volumes and the config map. The spec is used as the basis for starting the necessary pods and containers.
+    3. The containerized application executes. The expected mounts are available per the requirements and celebration occurs.
+
+Example
+-------
+
+Say I authored a simple application, `foo`, that requires Rabbit local GFS2 storage and a persistent Lustre storage volume. As the author, my program is coded to expect the GFS2 volume is mounted at `/foo/local` and the Lustre volume is mounted at `/foo/persistent`
+
+Working with an administrator, my applications storage requirements and pod specification are placed in an NNF Container Profile `foo`
+
+```yaml
+kind: NnfContainerProfile
+apiVersion: v1alpha1
+metadata:
+    name: foo
+    namespace: default
+spec:
+    storages:
+    - name: $JOB_DW_foo-local-storage
+      type: gfs2
+    - name: $PERSISTENT_DW_foo-persistent-storage
+      type: lustre
+    podSpec:
+        containers:
+        - name: foo
+          image: foo:latest
+          command:
+          - /foo
+          volumeMounts:
+          - name: foo-local-storage
+            mountPath: /foo/local
+          - name: foo-persistent-storage
+            mountPath: /foo/persistent
+```
+
+Say Peter wants to use `foo` as part of his job specification. Peter would submit the job with the below directives
+
```

Comment:
Not a fan of the $JOB_DW or $PERSISTENT_DW here.  In previous uses these were symbols that were like environment variables in the #DWs and actual environment variables in the process executing the user's job script.  In your proposal, they aren't anything like an environment variable and yet look like them.  In the very least, in this particular context, drop the dollar sign (having trouble escaping it properly in this comment, \\$).

With the proposal there are 4 things that look similar but are actually quite different:
1. JOB_DW_my-gfs2
2. JOB_DW_foo-local-storage
3. PERSISTENT_DW_some-lustre
4. PERSISTENT_DW_foo-persistent-storage

I'm guessing you are trying to accommodate the DWS generic parser with this proposal but that's kind of the tail wagging the dog here, i.e., what is best for users should win out over what the parser is capable of.  If you keep the key in #DW container equal to what gets put in to /spec/volumes[] then it's clear/consistent how what is in the #DW connects to the /spec/volumes[] which in turn connects to the /spec/containers[]/volumeMounts[].

You could change things to
`#DW container ... foo-local-storage=JOB_DW_my-gfs2 foo-persistent-storage=PERSISTENT_DW_some-lustre`
which would at least eliminate the two new variants of JOB_DW/PERSISTENT_DW.

Reply:
Will drop the $